akka-http-core/src/main/scala/akka/http/impl/engine/ws/WebSocket.scala

```diff
@@ -89,6 +90,21 @@ private[http] object WebSocket {
   private[this] final val PongFullFrame: FrameStart = FrameEvent.fullFrame(Opcode.Pong, None, ByteString.empty, fin = true)
   private[this] final val mkDirectAnswerPong = () => DirectAnswer(PongFullFrame)
 
+  /** The layer that transparently monitors data flow in both directions and fails the flow
+   *  with [[akka.stream.StreamIdleTimeoutException]] if elements are not passed within configured time.
+   *  */
+  def idleTimeout(settings: WebSocketSettings): BidiFlow[FrameHandler.Output, FrameHandler.Output, FrameOutHandler.Input, FrameOutHandler.Input, NotUsed] = {
+    val receiveIdleTimeoutFlow = settings.receiveIdleTimeout match {
+      case receiveIdleTimeout: FiniteDuration => Flow[FrameHandler.Output].idleTimeout(receiveIdleTimeout)
+      case _ => Flow[FrameHandler.Output].map(identity)
```

Comment:
Just `Flow[FrameHandler.Output]` already is an identity flow, so need for the additional map, same for the input one when disabled.